drivers/temperature/max31865pmb1/max31865.c

```diff
@@ -0,0 +1,361 @@
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+
+#include <errno.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <stdbool.h>
+#include "max31865.h"
+#include "no_os_spi.h"
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+/******************************************************************************/
+
+/**
+ * @brief Device and comm init function
+ * @param device - MAX31865 descriptor to be initialized
+ * @param init_param - Init parameter for descriptor
+ * @return 0 in case of success, errno errors otherwise
+ */
+int max31865_init(struct max31865_dev **device,
+		  struct max31865_init_param *init_param)
+{
+	int ret;
+	struct max31865_dev *descriptor;
+
+	descriptor = (struct max31865_dev *)calloc(1, sizeof(*descriptor));
+	if (!descriptor)
+		return -ENOMEM;
+
+	ret = no_os_spi_init(&descriptor->comm_desc, &init_param->spi_init);
+
+	if (ret){
+		free(descriptor);
+		return ret;
+	}
+
+	*device = descriptor;
+
+	return 0;
+}
+
+/**
+ * @brief Remove resources allocated by the init function
+ * @param device  - max31865 descriptor
+ * @return 0 in case of success, errno errors otherwise
+ */
+int max31865_remove(struct max31865_dev *device)
+{
+	int ret;
+	if (!device)
+		return -ENODEV;
+
+	ret = no_os_spi_remove(device->comm_desc);
+	if (ret)
+		return -EINVAL;
+
+	free(device);
+
+	return 0;
+}
+
+/**
+ * @brief Read raw register value
+ * @param device - MAX31865 descriptor
+ * @param val - register value
+ * @return 0 in case of success, negative error code otherwise
+ */
+int max31865_read_raw(struct max31865_dev *device, uint8_t *val)
```

Comment:
please use standard function prototype for the read/write functions. see an example:
https://github.com/analogdevicesinc/no-OS/blob/d6ca269bc4b06260c18a1a767f00678a284f57a5/drivers/adc/adaq8092/adaq8092.h#L231-L236